includes/class-wc-payment-gateway-wcpay.php

```diff
@@ -1224,6 +1224,8 @@ public function process_payment_for_order( $cart, $payment_information, $schedul
 			}
 
 			if ( empty( $intent ) ) {
+				$this->duplicate_payment_prevention_service->remove_and_cancel_attached_intent_id( $order );
```

Comment:
Since this is the part of the checkout flow, I think it would be wise to cover this flow with the tests. 
On another hand, did we check if attached payment intent is already captured?  In that case, wouldn't this cause an exception?